primitives/state-machine/src/ext.rs

```diff
@@ -354,17 +327,21 @@ where
 	}
 
 	fn place_storage(&mut self, key: StorageKey, value: Option<StorageValue>) {
-		trace!(target: "state", "{:04x}: Put {}={:?}",
-			self.id,
-			HexDisplay::from(&key),
-			value.as_ref().map(HexDisplay::from)
-		);
 		let _guard = guard();
 		if is_child_storage_key(&key) {
 			warn!(target: "trie", "Refuse to directly set child storage key");
 			return;
 		}
 
+		let ready_to_encode_result = value.as_ref().map(|v| EncodeOpaqueValue(v.clone()));
```

Comment:
Same question as above - does this incur a cost when not tracing?

Reply:
yeah, I think so move into the the `trace! block` and it will only included when tracing/trace is enabled AFAIU.